File: payment-processor/src/payment/api/dtos/payment-process.request.ts
import { IsCPF } from '@infra/common/decorators/validation/is-cpf.decorator';
import { IsEnum, IsNotEmpty, IsNumber, IsUUID, Min } from 'class-validator';
import { PaymentMethod } from 'src/payment/domain/enums/payment-method.enum';

export class PaymentProcessRequest {
  @IsEnum(PaymentMethod)
  paymentMethod: PaymentMethod;

  @IsUUID()
  orderId: string;

  @IsUUID()
  itemId: string;

  @IsNumber()
  @Min(1)
  amount: number;

  @IsNotEmpty()
  buyerName: string;

  @IsNotEmpty()
  @IsCPF()
  buyerCpf: string;
}

File: orders-service/src/main/java/com/rpe/challenge/payments/integration/clients/payment/dtos/PaymentProcessRequest.java
package com.rpe.challenge.payments.integration.clients.payment.dtos;


import com.rpe.challenge.orders.domain.enums.PaymentMethod;

import java.util.UUID;

public record PaymentProcessRequest(
	PaymentMethod paymentMethod,
	UUID orderId,
	UUID itemId,
	long amount,
	String buyerName,
	String buyerCpf
) {
}
